indexer/packages/postgres/__tests__/stores/wallet-table.test.ts

```diff
@@ -90,48 +90,68 @@ describe('Wallet store', () => {
-
-    const persistentCache = await PersistentCacheTable.findById('totalVolumeUpdateTime');
-    const lastUpdateTime = persistentCache?.value
-      ? DateTime.fromISO(persistentCache.value)
-      : undefined;
+  describe('Wallet .updateTotalVolume()', () => {
+    it('Successfully updates totalVolume for time window multiple times', async () => {
+      const firstFillTime: DateTime = await populateWalletSubaccountFill();
+
+      // Update totalVolume for a time window that covers all fills
+      await WalletTable.updateTotalVolume(
+        firstFillTime.minus({ hours: 1 }).toISO(), // need to minus because left bound is exclusive
+        firstFillTime.plus({ hours: 1 }).toISO(),
+      );
+      let wallet: WalletFromDatabase | undefined = await WalletTable
+        .findById(defaultWallet.address);
+      expect(wallet).toEqual(expect.objectContaining({
+        ...defaultWallet,
+        totalVolume: '103',
+      }));
+
+      // Update totalVolume for a time window that excludes some fills
+      // For convenience, we will reuse the existing fills data. The total volume calculated in this
+      // window should be added to the total volume above.
+      await WalletTable.updateTotalVolume(
+        firstFillTime.toISO(), // exclusive -> filters out first fill from each subaccount
+        firstFillTime.plus({ minutes: 2 }).toISO(),
+      );
+      wallet = await WalletTable.findById(defaultWallet.address);
+      expect(wallet).toEqual(expect.objectContaining({
+        ...defaultWallet,
+        totalVolume: '105', // 103 + 2
+      }));
+    });
 
-    expect(lastUpdateTime).not.toBeUndefined();
-    if (lastUpdateTime?.toMillis() !== undefined) {
-      expect(lastUpdateTime.toMillis()).toEqual(rightBound.toMillis());
-    }
+    it('Successfully upserts persistent cache', async () => {
+      const referenceDt = DateTime.utc();
+
+      // Sets initial persistent cache value
+      let leftBound: DateTime = referenceDt.minus({ hours: 2 });
+      let rightBound: DateTime = referenceDt.minus({ hours: 1 });
+
+      await WalletTable.updateTotalVolume(leftBound.toISO(), rightBound.toISO());
+
+      let persistentCache: PersistentCacheFromDatabase | undefined = await PersistentCacheTable
+        .findById(PersistentCacheKeys.TOTAL_VOLUME_UPDATE_TIME);
+      let lastUpdateTime: string | undefined = persistentCache?.value;
+      expect(lastUpdateTime).not.toBeUndefined();
+      if (lastUpdateTime !== undefined) {
+        expect(lastUpdateTime).toEqual(rightBound.toISO());
+      }
+
+      // Updates persistent cache value
+      leftBound = referenceDt.minus({ hours: 1 });
+      rightBound = referenceDt;
+
+      await WalletTable.updateTotalVolume(leftBound.toISO(), rightBound.toISO());
+
+      persistentCache = await PersistentCacheTable.findById(
+        PersistentCacheKeys.TOTAL_VOLUME_UPDATE_TIME,
+      );
+      lastUpdateTime = persistentCache?.value;
```

Comment:
The preference in general is to use const and keep objects immutable. Constantly overwriting them is an antipattern and can be confusing. When possible please use `const` instead of `let` even in tests

Reply:
bump on this